hack/jenkins/upload-to-gcs.sh

```diff
@@ -151,15 +151,22 @@ function update_job_result_cache() {
 
   mkdir -p ${tmp_results%/*}
 
+  # Construct a valid json file
+  echo "{\"JOB\" : [" > ${tmp_results}
+
   for upload_attempt in $(seq 3); do
     echo "Copying ${job_results} to ${tmp_results} (attempt ${upload_attempt})"
-    gsutil -q cp ${job_results} ${tmp_results} 2>&- || continue
+    gsutil -q cat ${job_results} 2>&- |\
+     sed -n 's/^\({"version".*}\),*/\1,/p' >> ${tmp_results} || continue
+
     break
   done
 
   echo "{\"version\": \"${version}\", \"buildnumber\": \"${BUILD_NUMBER}\"," \
        "\"result\": \"${build_result}\"}" >> ${tmp_results}
 
+  echo -e "{}\n]}" >> ${tmp_results}
```

Comment:
might be worth adding a comment "JSON won't allow the last element in an array to be followed by a comma" or something like that to explain the empty element.